components/match2/wikis/counterstrike/match_legacy.lua

```diff
@@ -0,0 +1,245 @@
+
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper', {requireDevIfEnabled = true})
+
+local LOSER_STATUSES = {'FF', 'DQ', 'L'}
+
+local MatchLegacy = {}
+
+function MatchLegacy.storeMatch(match2)
+	local match = MatchLegacy.convertParameters(match2)
+
+	match.games = MatchLegacy.storeGames(match, match2)
+
+	MatchLegacy.storeMatchSMW(match)
+
+	return mw.ext.LiquipediaDB.lpdb_match(
+		"legacymatch_" .. match2.match2id,
+		match
+	)
+end
+
+function MatchLegacy.convertParameters(match2)
+	local match = Table.deepCopy(match2)
+	for key, _ in pairs(match) do
+		if String.startsWith(key, "match2") then
+			match[key] = nil
+		end
+	end
+
+	if match.walkover == "ff" or match.walkover == "dq" then
+		match.walkover = match.winner
+	elseif match.walkover == "l" then
+		match.walkover = nil
+	end
+
+	match.staticid = match2.match2id
+
+	-- Handle extradata fields
+	local extradata = Json.parseIfString(match2.extradata)
+
+	local isOverturned = Logic.readBool(extradata.overturned)
+	if isOverturned then
+		match.walkover = match.winner
+		match.resulttype = 'ff'
+	end
+
+	match.extradata = {
+		bestofx = tostring(match2.bestof),
+		timezone = '',
+		timezoneID = '',
+		matchsection = extradata.matchsection,
+		opponent1rounds = 0,
+		opponent2rounds = 0,
+		overturned = Logic.readBool(extradata.overturned) and '1' or '',
+		hidden = Logic.readBool(extradata.hidden) and '1' or '0',
+		featured = Logic.readBool(extradata.featured) and '1' or '0',
+		icondark = Variables.varDefault("tournament_icon_dark"),
+		team1icon = match2.match2opponents[1] and match2.match2opponents[1].icon or nil,
+		team2icon = match2.match2opponents[2] and match2.match2opponents[2].icon or nil,
+		maps =''
+	}
+
+	if extradata.status then
+		if extradata.status == 'cancelled' or extradata.status == 'canceled' then
+			match.extradata.cancelled = '1'
+		else
+			match.extradata.cancelled = ''
+		end
+	end
+
+	local maps = {}
+	for gameIndex, game in ipairs(match2.match2games or {}) do
+		local scores = ''
+		if type(scores) == "string" then
+			scores = Json.parse(game.scores)
+		end
+		match.extradata.opponent1rounds = match.extradata.opponent1rounds + (tonumber(scores[1] or '') or 0)
+		match.extradata.opponent2rounds = match.extradata.opponent2rounds + (tonumber(scores[2] or '') or 0)
+		match.extradata['vodgame' .. gameIndex] = game.vod
+		table.insert(maps, game.map)
+	end
+	match.extradata.maps = table.concat(maps, ',')
+
+	local bracketData = Json.parseIfString(match2.match2bracketdata)
+	if type(bracketData) == "table" and bracketData.type == "bracket" then
+		local headerName
+		if bracketData.header then
+			headerName = (DisplayHelper.expandHeader(bracketData.header) or {})[1]
+		end
+		if String.isEmpty(headerName) then
+			headerName = Variables.varDefault("match_legacy_header_name")
+		end
+		if String.isNotEmpty(headerName) then
+			match.header = headerName
+			Variables.varDefine("match_legacy_header_name", headerName)
```

Comment:
match.bracketData.inheritedheader?!
(might need Json.parseIfString)

Reply:
Tbh I didn't pay attention to this part, and I don't know what to do here.

Reply:
Then the entire thing can be removed. Though it does allow for the schedule (for brackets) to be used. Leave it up to you either to remove it or adjust to the simplified version

Reply:
As we don't use it will be removed.